bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/capability/MeasureCapability.java

```diff
@@ -0,0 +1,105 @@
+    private final Logger logger = LoggerFactory.getLogger(MeasureCapability.class);
+    private final Map<String, State> measures = new HashMap<>();
+    private final MeasuresChannelHelper measureChannelHelper;
+
+    public MeasureCapability(CommonInterface handler, List<ChannelHelper> helpers) {
+        super(handler);
+        measureChannelHelper = (MeasuresChannelHelper) helpers.stream().filter(c -> c instanceof MeasuresChannelHelper)
+                .findFirst().orElseThrow(() -> new IllegalArgumentException(
+                        "MeasureCapability must find a MeasuresChannelHelper, file a bug."));
+    }
+
+    @Override
+    public void initialize() {
+        ApiBridgeHandler bridgeApi = handler.getRootBridge();
+        if (bridgeApi != null) {
+            api = Optional.ofNullable(bridgeApi.getRestManager(WeatherApi.class));
+            measureChannelHelper.setMeasures(measures);
+        }
+    }
+
+    @Override
+    public List<NAObject> updateReadings(WeatherApi api) {
+        String bridgeId = handler.getBridgeId();
+        String deviceId = bridgeId != null ? bridgeId : handler.getId();
+        String moduleId = bridgeId != null ? handler.getId() : null;
+        updateMeasurements(api, deviceId, moduleId);
+        return List.of();
+    }
+
+    private void updateMeasurements(WeatherApi api, String deviceId, @Nullable String moduleId) {
+        measures.clear();
+        thing.getChannels().stream().filter(channel -> !channel.getConfiguration().getProperties().isEmpty())
+                .forEach(channel -> {
+                    ChannelTypeUID channelTypeUID = channel.getChannelTypeUID();
+                    if (channelTypeUID != null) {
+                        MeasureConfiguration measureDef = channel.getConfiguration().as(MeasureConfiguration.class);
+                        String apiDescriptor = channelTypeUID.getId().split("-")[0];
+                        try {
+                            Object result = measureDef.limit.isBlank()
+                                    ? api.getMeasurements(deviceId, moduleId, measureDef.period, apiDescriptor)
+                                    : api.getMeasurements(deviceId, moduleId, measureDef.period, apiDescriptor,
+                                            measureDef.limit);
+                            MeasureClass.AS_SET.stream().filter(mt -> mt.apiDescriptor.equals(apiDescriptor))
+                                    .findFirst().ifPresent(mt -> {
+                                        State data = result instanceof ZonedDateTime
+                                                ? toDateTimeType((ZonedDateTime) result)
+                                                : result instanceof Double ? toQuantityType((Double) result, mt)
+                                                        : UnDefType.UNDEF;
+                                        measures.put(channel.getUID().getIdWithoutGroup(), data);
+                                    });
```

Comment:
Minor: I would suggest to rename mt into mc and data into state

Reply:
Some cleanup done here to have it more readable.